oktoberfest/data/spectra.py

```diff
@@ -148,36 +178,54 @@ def get_columns(self, fragment_type: FragmentType, return_column_names: bool = F
         """
         prefix = Spectra._resolve_prefix(fragment_type)
         logger.debug(prefix)
-        columns_to_select = list(filter(lambda c: c.startswith(prefix), self.spectra_data.columns))
-        if return_column_names:
-            return scipy.sparse.csr_matrix(self.spectra_data[columns_to_select].values), columns_to_select
+
+        if fragment_type.value == 1:
+            matrix = scipy.sparse.csr_matrix(self.spectra_data.layers["pred_int"].toarray())
+        elif fragment_type.value == 2:
+            matrix = scipy.sparse.csr_matrix(self.spectra_data.layers["raw_int"].toarray())
+        else:
+            matrix = scipy.sparse.csr_matrix(self.spectra_data.X.toarray())
+
         # Check if conversion is low change to coo then csr from coo
-        return self.spectra_data[columns_to_select]
+        return matrix, self._gen_column_names(fragment_type) if return_column_names else matrix
 
     def get_matrix(self, fragment_type: FragmentType) -> Tuple[csr_matrix, List[str]]:
         """
-        Get intensities sparse matrix from dataframe.
+        Get intensities sparse matrix from AnnData object.
 
         :param fragment_type: choose predicted, raw, or mz
         :return: sparse matrix with the required data
         """
         prefix = Spectra._resolve_prefix(fragment_type)
         logger.debug(prefix)
-        columns_to_select = list(filter(lambda c: c.startswith(prefix), self.spectra_data.columns))
-        return scipy.sparse.csr_matrix(self.spectra_data[columns_to_select].values), columns_to_select
+        if fragment_type.value == 1:
+            matrix = self.spectra_data.layers["pred_int"]
+        elif fragment_type.value == 2:
+            matrix = self.spectra_data.layers["raw_int"]
+        else:
+            matrix = self.spectra_data.X
+        # Check if conversion is low change to coo then csr from coo
+        return matrix, self._gen_column_names(fragment_type)
 
-    def write_as_hdf5(self, output_file: Union[str, Path]) -> Thread:
+    def write_as_hdf5(self, output_file: Union[str, Path], include_intensities: bool = False) -> Thread:
```

Comment:
please remove all the logic here. You can simply use anndata API here: anndata.write https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.write.html this is essentially a one liner. Also ensure gzip compression is active! Then also change the name of the generated output file in the documentation of oktoberfest in the docs subfolder.

Reply:
Throwing an error for now: `can't implicitly convert non-string objects to strings` for the INTENSITIES column. In this issue (https://github.com/scverse/scanpy/issues/1866) I found this: "If the column's dtype is object, we check to see if it's string values, otherwise we say we don't know how to write it, since it could be any mix of things." So not sure if its possible with AnnData write. I will look further into it.

Reply:
INTENSITIES can only be float! So if it says the dtype is object, then sth. is wrong.

Reply:
The INTENSITES were still in the obs column, so that caused the problem. I removed them and implemented the write function.